src/collectives/device/prims_ll.h

```diff
@@ -773,10 +773,12 @@ class Primitives<T, RedOp, Fan, Direct, ProtoLL, P2p>:
     return mscclGenericOp<0,1,0,0>(&srcs, 1, &dsts, 1, eltN);
   }
   __device__ void reduce(T** srcs, int nsrcs, T** dsts, int ndsts, int eltN) {
+#if !defined(__gfx942__)
```

Comment:
I don't think this is necessary since mscclRunInterpreter is templated. The "else" path of "Proto::Id == NCCL_PROTO_LL" should not have been included by compiler. Did you see any difference with and without this "#if"?

Reply:
You're right, I ran some experiments to compare and didn't find difference. Removed this in latest commit.